src/udev/udev-event.c

```diff
@@ -903,6 +903,13 @@ static int rename_netif(UdevEvent *event) {
         if (FLAGS_SET(flags, IFF_UP)) {
                 log_device_info(dev, "Network interface '%s' is already up, refusing to rename to '%s'.",
                                 oldname, event->name);
+
+                /* Since we cannot rename, set an alternative name so that the
+                 * new name can still be used for this interface. */
+                r = rtnl_set_link_alternative_names(&event->rtnl, ifindex, STRV_MAKE(event->name));
```

Comment:
This removes other alternative names.
At least, please obtain the current alternative names, and append the requested name.

Reply:
Based on my testing this does not seem to be the case. The kernel will just add this as a new alternative name, but not remove existing ones. Calling this with the current names results in `-EEXIST`.